core/domain/playthrough_issue_registry_test.py

```diff
@@ -18,14 +18,57 @@
 
 from __future__ import annotations
 
+from core import utils
 from core.domain import playthrough_issue_registry
 from core.tests import test_utils
+from extensions.issues.CyclicStateTransitions import CyclicStateTransitions
+from extensions.issues.EarlyQuit import EarlyQuit
+from extensions.issues.MultipleIncorrectSubmissions import (
+    MultipleIncorrectSubmissions)
 
 
 class IssueRegistryUnitTests(test_utils.GenericTestBase):
     """Test for the issue registry."""
 
+    def setUp(self):
+        super(IssueRegistryUnitTests, self).setUp()
+        self.issues_dict = {
+            'EarlyQuit': (
+                EarlyQuit.EarlyQuit),
+            'CyclicStateTransitions': (
+                CyclicStateTransitions.CyclicStateTransitions),
+            'MultipleIncorrectSubmissions': (
+                MultipleIncorrectSubmissions.MultipleIncorrectSubmissions)
+            }
+        self.invalid_issue_type = 'InvalidIssueType'
+
+    def tearDown(self):
+        super(IssueRegistryUnitTests, self).tearDown()
+        playthrough_issue_registry.Registry._issues = {} # pylint: disable=protected-access
+
     def test_issue_registry(self):
         """Do some sanity checks on the issue registry."""
         self.assertEqual(
             len(playthrough_issue_registry.Registry.get_all_issues()), 3)
+
+    def test_correct_issue_registry_types(self):
+        """Tests issue registry for fetching of issue instances of correct
+        issue types.
+        """
+        for issue_type, instance in self.issues_dict.items():
+            self.assertIsInstance(
+                playthrough_issue_registry.Registry.get_issue_by_type(
+                    issue_type), instance)
+
+    def test_incorrect_issue_registry_types(self):
+        """Tests issue registry for raising error on passing incorrect issues
```

Comment:
Tests that an error is raised when fetching an incorrect issue type.

Reply:
sure